src/main/scala/io/iohk/praos/domain/BlockSigner.scala

```diff
@@ -0,0 +1,22 @@
+package io.iohk.praos.domain
+
+import akka.util.ByteString
+import io.iohk.praos.crypto._
+
+
+object BlockSigner {
+  def signedWith(block: Block, privateKey: Key): Block = {
```

Comment:
I'm not sure whether having to pass a block with a fake signature to this function. Maybe we could have both `UnsignedBlock` and `Block` classes, what do you think?

Reply:
Honestly I don't like having an "empty" or "fake" signature for not-yet-signed blocks either. However, if you convience Alan V to use inheritance to subclass Block from a new UnsignedBlock class, I'll agree with you :)

Reply:
I don't like that either :stuck_out_tongue:
I was thinking that maybe we could have a `Block` contain an `UnsignedBlock`, WDYT?

Reply:
Yes, but we would basically be back to the point of having the Signed trait :)

Reply:
After discussing this with Nico and Alan V, we will revert these changes and stick to the `UnsignedBlock`/`Block`/`Signed[T]` modeling as defined by the Praos Formalization.